src/Lib.P2P/PubSub/FloodRouter.cs

```diff
@@ -0,0 +1,316 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Lib.P2P.Protocols;
+using ProtoBuf;
+using Semver;
+
+namespace Lib.P2P.PubSub
+{
+    /// <summary>
+    ///   The original flood sub router.
+    /// </summary>
+    public class FloodRouter : IPeerProtocol, IMessageRouter
+    {
+        private static ILog _log = LogManager.GetLogger(typeof(FloodRouter));
+
+        private MessageTracker _tracker = new MessageTracker();
+        private ConcurrentDictionary<string, string> _localTopics = new ConcurrentDictionary<string, string>();
+
+        /// <summary>
+        ///   The topics of interest of other peers.
+        /// </summary>
+        public TopicManager RemoteTopics { get; set; } = new TopicManager();
+
+        /// <inheritdoc />
+        public event EventHandler<PublishedMessage> MessageReceived;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused event 'MessageReceived' or invoke it.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)